Invalid generated C code with map with optional value

**V version:** V 0.3.1 267e26b
**OS:** Linux

**What did you do?**

I tried to use `map[string]?string`.

`map[string]string` instead works fine.


```
module main

import os

// android device
struct AdbDevice {
	id string
	opts map[string]?string
}

fn main() {
}
```

**What did you expect to see?**
No compile error
 
**What did you see instead?**
```
❯ v -cg os-import.v 
os-import.v:3:8: warning: module 'os' is imported but never used
    1 | module main
    2 | 
    3 | import os
      |        ~~
    4 | 
    5 | // android device
/tmp/v_1000/os-import.11999742646382012615.tmp.c:13277: error: cannot convert 'struct string' to 'unsigned char'
builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -3757,7 +3757,7 @@ fn (mut g Gen) unlock_locks() {
 
 fn (mut g Gen) map_init(node ast.MapInit) {
 	unwrap_key_typ := g.unwrap_generic(node.key_type)
-	unwrap_val_typ := g.unwrap_generic(node.value_type)
+	unwrap_val_typ := g.unwrap_generic(node.value_type).clear_flag(.optional).clear_flag(.result)
 	key_typ_str := g.typ(unwrap_key_typ)
 	value_typ_str := g.typ(unwrap_val_typ)
 	value_sym := g.table.sym(unwrap_val_typ)

--- a/vlib/v/gen/c/index.v
+++ b/vlib/v/gen/c/index.v
@@ -378,7 +378,7 @@ fn (mut g Gen) index_of_map(node ast.IndexExpr, sym ast.TypeSymbol) {
 	elem_type_str := if elem_sym.kind == .function {
 		'voidptr'
 	} else {
-		g.typ(elem_type)
+		g.typ(elem_type.clear_flag(.optional).clear_flag(.result))
 	}
 	get_and_set_types := elem_sym.kind in [.struct_, .map]
 	if g.is_assign_lhs && !g.is_arraymap_set && !get_and_set_types {

--- a/vlib/v/tests/map_value_with_optional_result_test.v
+++ b/vlib/v/tests/map_value_with_optional_result_test.v
@@ -1,0 +1,11 @@
+import os
+
+struct AdbDevice {
+	opts map[string]?string
+}
+
+fn test_map_value_with_optional_result() {
+	// void warnings
+	_ := os.max_path_len
+	assert true
+}

--- a/vlib/v/tests/map_value_with_optional_result_test.v
+++ b/vlib/v/tests/map_value_with_optional_result_test.v
@@ -5,7 +5,7 @@ struct AdbDevice {
 }
 
 fn test_map_value_with_optional_result() {
-	// void warnings
+	// avoid warnings
 	_ := os.max_path_len
 	assert true
 }

